### PR TITLE
Implement SkipFinalSnapshot on delete

### PR DESCRIPTION
**Pull request by liskl**
- update for FinalSnapshot Functionality in RDS, thanks @amazzalel-habib
- update to go v1.17
- updated dependencies

### CODE REVIEW

**Comment by liskl** [comment]
@sorenmat would you be able to review this?

File: main.go
Comment (sorenmat) [inline main.go:148]: Why create a new context?

Comment (liskl) [reply main.go]: Sorry for that reverted

**Comment by liskl** [comment]
Sorry for the wasted time on the first one, I'll get better I swear, care to try for a second review @sorenmat ?

**Review by sorenmat — APPROVED** [review]
LGTM

**Pull request merged**
